pkg/kudoctl/util/repo/repo_operator.go

```diff
@@ -79,21 +79,43 @@ func (r *Client) DownloadIndexFile() (*IndexFile, error) {
 }
 
 // getPackageReaderByFullPackageName downloads the tgz file from the remote repository and unmarshals it to the package CRDs
-func (r *Client) getPackageReaderByFullPackageName(fullPackageName string) (io.Reader, error) {
-	var fileURL string
-	parsedURL, err := url.Parse(r.Config.URL)
+// this will cycle through urls for the
+func (r *Client) getPackageReaderByFullPackageName(pkg *PackageVersion) (io.Reader, error) {
+
+	pkgURL, err := repositoryBackedPkgURL(r.Config.URL, pkg)
 	if err != nil {
-		return nil, errors.Wrap(err, "parsing config url")
+		return nil, err
 	}
-	parsedURL.Path = fmt.Sprintf("%s/%s.tgz", parsedURL.Path, fullPackageName)
+	urls := append(pkg.URLs, pkgURL)
```

Comment:
ooh so this is the magic that actually uses the index file, it really took me a while to find where that happens.